src/main/cpp/DragonVision/DragonLimelight.cpp

```diff
@@ -486,10 +486,27 @@ units::length::inch_t DragonLimelight::EstimateTargetZDistance_RelToRobotCoords(
 {
     if (EstimateTargetZDistance().to<double>() != -1.0)
     {
-        units::length::inch_t targetZoffset_RF_inch = EstimateTargetZDistance() + m_mountingZOffset; ///< the offset is positive if the limelight is above the center of the robot
+        units::length::inch_t targetZoffset_RF_inch = EstimateTargetZDistance() + GetMountingZOffset(); ///< the offset is positive if the limelight is above the center of the robot
 
         return targetZoffset_RF_inch;
     }
     else
         return units::length::inch_t(-1.0);
 }
+
+VisionData DragonLimelight::GetPoseRelativeToApriltag()
+{
+    auto tagetpoes = m_networktable.get()->GetDoubleArrayTopic("targetpose_cameraspace");
+
+    std::vector<double> vector = tagetpoes.GetEntry(std::array<double, 6>{}).Get();
+    // targetpose_cameraspace returns distance and angle
+    // assign variables based on the vector, then construct a translation 3d
+    units::length::meter_t Xdist{vector[0]};
+    units::length::meter_t Ydist{vector[1]};
+    units::length::meter_t Zdist{vector[2]};
+    units::angle::degree_t Xangle{vector[3]};
+    units::angle::degree_t Yangle{vector[4]};
+    units::angle::degree_t Zangle{vector[5]};
+    auto translation = frc::Translation3d(Xdist, Ydist, Zdist);
+    return VisionData{translation};
```

Comment:
Can also return apriltag id by calling GetAprilTagId() for the second argument in VisionData

Reply:
fixed